src/tbb/scheduler_common.h

```diff
@@ -148,6 +149,10 @@ struct task_accessor {
     context execution **/
 template <bool report_tasks>
 class context_guard_helper {
+    // The number of unsuccessful attempts to retrieve a non-local task, after which ITT_TASK_END notification
+    // should be sent. If the notification is postponed for too long, data in profiling tools might get skewed.
+    static constexpr int itt_task_end_threshold = 2;
```

Comment:
What is the way to choose the best value for this threshold? The overhead of these calls are only seen if there is an ITT library successfully loaded, right? So we'd only see additional runtime overhead when traces are being actively being collected.

Reply:
That is correct - the extra overhead is only realized when a collector is active. As far as the thresholds go, I have tested with 0, 1, 2 and saw no difference in the test cases I ran. TBB is generally shown as active for longer with higher thresholds, so any analytics built to tune TBB algorithms will start to introduce error in the analysis. However, the differences are negligible from 0 to 2.

Reply:
If a threshold of 0 is ok, we don't need task_search_count at all.

Reply:
It would need to be tested on some heavily imbalanced workloads and/or producer/consumer patterns where stealing prevails. Without that, I think a low value like 2 is better,